src/main/java/command/CommandHandler.java

```diff
@@ -0,0 +1,178 @@
+            performTodo(userInput);
+        } else if (commandCode == Constant.INPUT_CODE_DEADLINE) {
+            performDeadline(userInput);
+        } else if (commandCode == Constant.INPUT_CODE_EVENT) {
+            performEvent(userInput);
+        } else if (commandCode == Constant.INPUT_CODE_DELETE) {
+            performDelete(userInput);
+        } else if (commandCode == Constant.INPUT_CODE_FIND) {
+            performFind(userInput);
+        } else if (commandCode == Constant.INPUT_CODE_INVALID) {
+            performInvalid();
+        } else if (commandCode == Constant.INPUT_CODE_DEFAULT_INVALID) {
+            performDefault();
+        } else {
+            System.out.println("Unknown Operation!");
+            System.out.println(Constant.DIVIDER_LINE);
+        }
+    }
+
+    /**
+     * Perform the list all task in task list operation.
+     */
+    private static void performList() {
+        System.out.println(Constant.DIVIDER_LINE);
+        Ui.printEntireCollection();
+        System.out.println(Constant.DIVIDER_LINE);
+    }
+
+    /**
+     * Perform the mark as done operation.
+     *
+     * @param userInput is the input from the console terminal.
+     */
+    private static void performDone(String userInput) {
+        System.out.println(Constant.DIVIDER_LINE);
+        int indexFromUserInput = Parser.getIndexFromUserInput(userInput);
+        if (indexFromUserInput > taskHandler.getTaskCount()) {
+            System.out.println("There is no task number " + indexFromUserInput + " to mark done.");
+            System.out.println("Please try again!");
+        } else if (taskHandler.checkStatus(indexFromUserInput - 1)) {
+            System.out.println("You have already completed this task.");
+        } else {
+            taskHandler.markDone(indexFromUserInput - 1);
+            System.out.println("Nice! I've marked the task as done:");
+            Ui.printTaskDetails(indexFromUserInput - 1);
+        }
+        System.out.println(Constant.DIVIDER_LINE);
+    }
+
+    /**
+     * Perform the todo operation by adding a todo task to the task list.
+     *
+     * @param userInput is the input from the console terminal.
+     */
+    private static void performTodo(String userInput) {
+        System.out.println(Constant.DIVIDER_LINE);
+        String taskDescription = Parser.extractTaskDescription(userInput);
+        taskHandler.addTask(new Todo(taskDescription));
+        Ui.printAddedTask(taskHandler.getTaskCount());
+        taskHandler.increaseTaskCount();
+        System.out.println(Constant.DIVIDER_LINE);
+    }
+
+    /**
+     * Perform the deadline operation by adding a deadline task to the task list.
+     *
+     * @param userInput is the input from the console terminal.
+     */
+    private static void performDeadline(String userInput) {
+        String taskDescription, taskTiming;
+        System.out.println(Constant.DIVIDER_LINE);
+        taskDescription = Parser.extractTaskDescription(userInput);
```

Comment:
quite a few  methods have these lines in common. you may want to extract it as a method by itself.
that improves both readability and SLAP

Reply:
Hi Professor Akshay, I am not too sure how to extract to further improve the SLAP design. My current implementation already extracted the information required via another method and only two methods, performDeadline and performEvent, use the extractTaskDescription and extractTaskTiming method in Parser class.
I tried to extract it into a method but eventually, my add task method still requires two arguments. (The only way is to extract the two arguments and return them into a string array)

May I ask how can I further improve my design? (Attached is my current attempt to improve the SLAP by reducing the string declaration line)
![image](https://user-images.githubusercontent.com/59407923/109390890-34ca9100-794f-11eb-965e-ec20133a15c9.png)